src/libsystemd/sd-event/sd-event.c

```diff
@@ -2051,6 +2052,32 @@ _public_ int sd_event_add_inotify(
         return 0;
 }
 
+_public_ int sd_event_add_inotify_fd(
+                sd_event *e,
+                sd_event_source **ret,
+                int fd,
+                uint32_t mask,
+                sd_event_inotify_handler_t callback,
+                void *userdata) {
+
+        int r;
+
+        r = sd_event_add_inotify(e, ret, FORMAT_PROC_FD_PATH(fd), mask, callback, userdata);
```

Comment:
Hmm, so this seems to be reversed in logic. We start with an fd, we convert it to a path, and then in sd_event_add_inotify() we convert it back to an fd. Why not make sd_event_add_inotify() a wrapper around sd_event_add_inotify_fd() instead?